be/src/modules/user/user.service.ts

```diff
@@ -159,4 +159,67 @@ export class UserService {
     await userRepository.update({ user_id: user_id }, userParam);
     return;
   }
+
+  async checkUserDeleteStatus(userId: string, entityManager?: EntityManager) {
+    const userRepository = entityManager
+      ? entityManager.getRepository<User>('user')
+      : this.userRepository;
+      
+    const deleteStatus = await userRepository.
+    createQueryBuilder('user')
+    .where("user.user_id = :id", { id: userId })
+    .andWhere('user.is_deleted = :is_deleted', { is_deleted: 1 })
+    .getOne();
+    
+    if(!deleteStatus){
+      return 0
+    }
+    else{
+      return 1
+    }
+  }
```

Comment:
Xem lại hàm này có cần thiết không?

Reply:
hàm này có cần thiết nhé, vì là mình xóa mềm chứ không phải xóa thẳng nên cần kiểm tra trước khi khóa hay mở khóa tài khoản.

Reply:
- `checkUserByUserId` thế vậy hàm này để làm gì?
- Nếu hàm này trả về 1 user sao ko sử dụng is_deleted luôn, ko cần `checkUserByUserId` nữa, đúng ko nhỉ?

Reply:
1 hàm sẽ là lấy ra tình trạng của việc xóa, 1 hàm kiểm tra xem người dùng đó
 có tồn tại trong db hay không. Hoàn toàn khác nhau.

Reply:
Nhưng mà sao không dùng chung cái `checkUserByUserId` luôn, nó cũng có `is_deleted` mà

Reply:
có luôn à, tôi không biết. Mà tôi nghĩ viết hàm sử dụng đúng mục đích để sau dễ dùng đi.

Reply:
- nó bị thừa thôi
- ok, vậy ông sửa lại đi rồi push lên đên tôi merge, nhớ push đúng branch ấy

Reply:
vẫ đang có conflict kìa :))

Reply:
bận gì không vào meet :))

Reply:
có việc gì vậy

Reply:
flow đúng cái tôi viết trên trello là được mà
có bước chọn code ý thì lúc đấy ông tự so sánh code main mả chọn